mobile/mobile.go

```diff
@@ -0,0 +1,612 @@
+		return err
+	}
+
+	log.Debugf("Mobile client connected to RPC")
+	return nil
+}
+
+// IsConnected returns whether or not there is an active connection.
+func IsConnected(namespace string) (bool, error) {
+	mMutex.Lock()
+	defer mMutex.Unlock()
+
+	mc, ok := m[namespace]
+	if !ok {
+		return false, fmt.Errorf("unknown namespace: %s", namespace)
+	}
+
+	return mc.lndConn != nil, nil
+}
+
+// Disconnect closes the RPC connection.
+func Disconnect(namespace string) error {
+	mMutex.Lock()
+	defer mMutex.Unlock()
+
+	mc, ok := m[namespace]
+	if !ok {
+		return fmt.Errorf("unknown namespace: %s", namespace)
+	}
+
+	if mc.lndConn != nil {
+		if err := mc.lndConn.Close(); err != nil {
+			log.Errorf("Error closing RPC connection: %v", err)
+		}
+		mc.lndConn = nil
+	}
+
+	return nil
+}
+
+// Status returns the status of the LNC RPC connection.
+func Status(namespace string) (string, error) {
+	mMutex.Lock()
+	defer mMutex.Unlock()
+
+	mc, ok := m[namespace]
+	if !ok {
+		return "", fmt.Errorf("unknown namespace: %s", namespace)
+	}
+
+	if mc.statusChecker == nil {
+		return "", nil
+	}
+
+	return mc.statusChecker().String(), nil
+}
+
+// RegisterLocalPrivCreateCallback sets up the native callbacks upon
+// creation of local private key.
+func RegisterLocalPrivCreateCallback(namespace string, c NativeCallback) error {
+	mMutex.Lock()
+	defer mMutex.Unlock()
+
+	mc, ok := m[namespace]
+	if !ok {
+		return fmt.Errorf("unknown namespace: %s", namespace)
+	}
+
+	mc.localPrivCreateCallback = c
+
+	return nil
+}
+
+// RegisterRemoteKeyReceiveCallback sets up the native callbacks upon
+// receiving the remote key from the server.
+func RegisterRemoteKeyReceiveCallback(namespace string, c NativeCallback) error {
+	mMutex.Lock()
+	defer mMutex.Unlock()
+
+	mc, ok := m[namespace]
+	if !ok {
+		return fmt.Errorf("unknown namespace: %s", namespace)
+	}
+
+	mc.remoteKeyReceiveCallback = c
+
+	return nil
+}
+
+// RegisterAuthDataCallback sets up the native callbacks upon
+// receiving auth data.
+func RegisterAuthDataCallback(namespace string, c NativeCallback) error {
+	mMutex.Lock()
+	defer mMutex.Unlock()
+
+	mc, ok := m[namespace]
+	if !ok {
+		return fmt.Errorf("unknown namespace: %s", namespace)
+	}
+
+	mc.authDataCallback = c
+
+	return nil
+}
+
+// InvokeRPC makes a synchronous RPC call.
+func InvokeRPC(namespace string, rpcName string, requestJSON string,
+	c NativeCallback) error {
+
+	mMutex.Lock()
+	defer mMutex.Unlock()
+
+	mc, ok := m[namespace]
+	if !ok {
+		return fmt.Errorf("unknown namespace: %s", namespace)
+	}
+
+	if rpcName == "" {
+		return fmt.Errorf("param rpcName required")
+	}
+
+	if requestJSON == "" {
+		return fmt.Errorf("param requestJSON required")
+	}
+
+	if mc.lndConn == nil {
+		return fmt.Errorf("RPC connection not ready")
+	}
+
+	method, ok := registry[rpcName]
+	if !ok {
+		return fmt.Errorf("rpc with name " + rpcName + " not found")
+	}
+
+	go func() {
+		log.Infof("Calling '%s' on RPC with request %s", rpcName, requestJSON)
```

Comment:
nit: wrap at 80

Reply:
this one's 73

Reply:
You need to set your IDE to use 8 spaces for each tab character. Same for the comment below.